### PR TITLE
Fix unpredictable choice between methods when using method binding.

### LINKED ISSUE #5076: Unpredictable choice between method overloads when binding to command
#### Example
XAML part:
```
...
<Button Command="{Binding MyHandler}" />
...
```
Code-behind:
```
...
// [1]
public void MyHandler(string x) {}
// [2]
public void MyHandler() {}
...
```
Expected behavior: [2] is executed
Actual behavior: [1] is executed

If we will swap methods [1] and [2], then [2] will be executed.

Also, the problem appears when `CommandParameter` is specified: the expected behavior is that [1] will be executed always, but actually, it depends on the order of methods.

Bonus: `CommandParameter="zzz"` leads to app termination after button click when the chosen overload receives `int` parameter.

### PR DESCRIPTION
**Pull request by Takoooooo**
## What does the pull request do?
The idea is to only allow the usage in method bindings of methods without any parameters or methods with the parameter type of object. It is possible to decide which overload will be called at runtime but it will cause performance penalties which we want to avoid, currently, we find the correct method in the compile time which is obviously faster.
The complete idea is described by @grokys  in [this](https://github.com/AvaloniaUI/Avalonia/issues/5076#issuecomment-1238274606) comment.

## Breaking changes
This PR will introduce breaking changes in user code because currently we chose the first method in class that has the same name as in the binding. After this PR we chose the method with the parameter of type "object" first,then method without parameters.

## Fixed issues
Fixes #5076

### CODE REVIEW

File: src/Avalonia.Base/Data/Core/Plugins/MethodAccessorPlugin.cs [hunk 1]
```diff
@@ -55,13 +55,20 @@ public class MethodAccessorPlugin : IPropertyAccessorPlugin
...
+                    var parameters = methodInfo.GetParameters();
+                    if (parameters.Length == 1 && parameters[0].ParameterType == typeof(object))
+                    {
+                        found = methodInfo;
+                        break;
+                    }
+                    else if (parameters.Length == 0)
+                    {
+                        found = methodInfo;
+                    }
```
Comment (ahopper) [inline src/Avalonia.Base/Data/Core/Plugins/MethodAccessorPlugin.cs:71]: should there be a break; here

Comment (jp2masa) [reply src/Avalonia.Base/Data/Core/Plugins/MethodAccessorPlugin.cs:71]: I don't think so, that would keep the bug, as the 1 parameter overload is preferred.

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0023662-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Review by maxkatz6 — APPROVED** [review]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0023687-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Pull request merged**